variod.c

```diff
@@ -268,18 +327,20 @@ int main(int argc, char *argv[])
 		
 	while(1) {
 		//Accept and incoming connection
-		fprintf(fp_console,"Waiting for incoming connections...\n");
+		fprintf(fp_console,"Waiting for incoming connections after %d sentences from sensors ...\n",
+			sensor_sentence_count);
 		fflush(fp_console);
-		c = sizeof(struct sockaddr_in);
 
+		c = sizeof(struct sockaddr);
 		//accept connection from an incoming client
-		connfd = accept(listenfd, (struct sockaddr *)&s_xcsoar, (socklen_t*)&c);
+		connfd = accept(listenfd, (struct sockaddr *)&sensor_server, (socklen_t*)&c);
```

Comment:
Accept initializes s_xcsoar, don´t see the need for complicating this that worked OK adding sensor_server.

Reply:
It's confusing to sax "s_xcsoar" when you mean the sockaddr  to sensord.
But more importantly, I initialized sin_zero field
Line 244:
`memset(&sensor_server,0,sizeof(sensor_server));`
That determines which connections it'g going to accept. (That's how I understand the documentation.)

Reply:
accept() and initializes s_xcsoar with the peer information.
Which peer? The first available in the queue.
Don't know how it may behave if you initialize it beforehand. Rewriting or error.

> int accept(int socket, struct sockaddr *address, int *address_len);
> 
> address
> The socket address of the connecting client that is filled in by accept() before it returns. The format of address is determined by the domain that the client resides in. This parameter can be NULL if the caller is not interested in the client address.
> 
> 

Taken from here, where it is explained clearly, MAN pages are confusing about this definition.
https://www.ibm.com/support/knowledgecenter/SSLTBW_2.2.0/com.ibm.zos.v2r2.bpxbd00/accept.htm